### DISCUSSION TITLE
How to requeue a job if failed with Nexrender Server, Worker & API

### DISCUSSION THREAD

**Discussion by Marcuzzz** [inlife/nexrender#766] [Q&A]
Question about Nexrender Server:

I'm not sure If I'm doing it right but:

I've created and run these Nexrender "services":
node cli/server.js
node cli/worker.js
node cli/api.js

api.js sends a job to the server.
a worker picks it up.

I've copied @nexrender/worker to another folder so I can make changes and add my own functionality to it.

For now, I only updated src/index.js "the do function"

```
 const client = createClient({ host, secret });

    do {
        let job = await nextJob(client, settings); {

            if (fs.existsSync(job.template.src)) { 
                settings.workpath = job.settings.workpath;
                settings.skipCleanup = job.settings.skipCleanup;
                settings.skipRender = job.settings.skipRender;
                job.state = 'started';
                job.startedAt = new Date()
            } else {
                job.state = 'requeue';
                job.error = 'Project can not found at' + job.template.src;
                job.errorAt = new Date();
            }
        }
...
```

api.js
```
const { createClient } = require('@nexrender/api')

const client = createClient({
    host: 'http://127.0.0.1:3000',
    secret: 'myapisecret',
})

const homedir = require('os').homedir();
const project = "path-to-job.json"
const job = require(project);

const main = async () => {

    const result = await client.addJob(job)

    result.on('created', job => console.log('project has been created'))
    result.on('started', job => console.log('project rendering started'))
    result.on('progress', (job, percents) => console.log('project is at: ' + percents + '%'))
    result.on('finished', job => console.log('project rendering finished'))
    result.on('error', err => console.log('project rendering error', err))
}

main().catch(console.error);

     
```

I would want the job to fail if for instance the AE project can't be found and I would like to automatically requeue the job and for the current worker I would like to ignore the current job that was created with the API... 

Any thoughts on how I could get started?
Basic question:
Should I requeue the job with the worker or should I requeue it with a new API request with api.js?
Can I get a list of workers from the server?

Any other documentation besides the gitrepo?

Any idea or help is much appreciated!
Cheers

**Comment by inlife** [comment]
The worker listing was a plan before, I haven't had a chance to work on that yet. If you wish to contribute, it would be great!!

I think theoretically, there is an option to have a dedicated API client that would be just monitoring jobs on the server. As soon as he sees a fail, he can immediately post a new one with the same parameters.

**Reply by Su2b0** [reply]
No, No, No, Inlife. This would cause hell, expecially when using it programatically. It should atleast be an option instead of a replacement.

We need a dedicated queue system.

**Reply by inlife** [reply]
I was talking about a re-queuing system for @Marcuzzz's specific case. Not as a part of nexrender, at least not right now

**Reply by Su2b0** [reply]
Ah right got it, my apologies!
